service.subtitles.torec/resources/lib/TorecSubtitlesDownloader.py

```diff
@@ -1,165 +1,244 @@
         santized_name = self.sanitize(movie_name)
-        log(__name__ , "Searching for %s" % santized_name)
-        subtitlePage = self.search_by_movie_name(santized_name)
-        if subtitlePage is None:
-            log(__name__ ,"Couldn't find relevant subtitle page")
+        log(__name__, "Searching for %s" % santized_name)
+        subtitle_page = self.search_by_movie_name(santized_name)
+        if subtitle_page is None:
+            log(__name__, "Couldn't find relevant subtitle page")
             return None
         else:
-            log(__name__ , "Found relevant meta data")
-            return subtitlePage
+            log(__name__, "Found relevant meta data")
+            return subtitle_page
 
     def search_by_movie_name(self, movie_name):
-        response = self.urlHandler.request("%s/ssearch.asp" % self.BASE_URL, {"search" : movie_name})
-        match = re.search('sub\.asp\?sub_id=(\w+)', response.data)
-        if (match is None):
+        """
+        Search for movie subtitle
+
+        :param movie_name:
+        :return:
+        """
+        response = self.opener.open(
+            "%s/ssearch.asp" % self.BASE_URL,
+            urllib.urlencode({"search": movie_name})
+        )
+        data = response.read()
+        match = re.search('sub\.asp\?sub_id=(\w+)', data)
+        if not match:
             return None
-          
-        subtitle_id = match.groups()[0]
-        subtitle_url = "%s/%s%s" % (self.BASE_URL, self.SUBTITLE_PATH, subtitle_id)
-        subtitle_data = self.urlHandler.request(subtitle_url).data
-        return SubtitlePage(subtitle_id, movie_name, subtitle_url, subtitle_data)
-        
-    def _requestSubtitle(self, subID, subURL):
-        params = {"sub_id"  : subID, 
-                  "s"       : 1440}
-                  
-        return self.urlHandler.request("%s/ajax/sub/guest_time.asp" % self.BASE_URL, params, 
-                                        ajax=True, referer=subURL, cookie=self._buildDefaultCookie(subID)).data
-        
-    def getDownloadLink(self, subID, optionID, subURL, persist=True):        
-        requestID = self._requestSubtitle(subID, subURL)
+
+        id_ = match.groups()[0]
+        sub_url = "%s/%s%s" % (self.BASE_URL, self.SUBTITLE_PATH, id_)
+        subtitle_data = self.opener.open(sub_url)
+        subtitle_data = subtitle_data.read()
+        return SubtitlePage(id_, movie_name, sub_url, subtitle_data)
+
+    def get_download_link(self, sub_id, option_id, persist=True):        
+        response = None
+        data = None
         
-        params = {"sub_id" : subID, "code": optionID, "sh" : "yes", "guest" : requestID, "timewated" : "15"}
+        params = {
+            "sub_id":     sub_id,
+            "code":       option_id,
+            "sh":         "yes",
+            "guest":      self._request_subtitle(sub_id),
+            "timewaited": "11",
+            "userAuth":   self._get_user_auth()
+        }
+
         for i in xrange(16):
```

Comment:
thanks @phil65, your comment actually made realize that the retry mechanism is no longer needed for this feature, so I removed it altogether.